src/components/ec-table/ec-table.vue

```diff
@@ -71,80 +71,80 @@
 
 import EcTableFooter from '../ec-table-footer';
 import EcTableHead from '../ec-table-head';
+import type { StickyColumnPosition, TableHeadColumn, TableHeadSort } from '../ec-table-head/types';
+import type { TableEvent, TableEvents } from './types';
 
 const isInCustomRowThreshold = useMediaQuery('(max-width: 768px)');
 const slots = useSlots();
 const attrs = useAttrs();
 // const emit = defineEmits(['sort', 'row-click']);
-const emit = defineEmits(['sort']);
-
-const props = defineProps({
-  columns: {
-    type: Array,
-    default: () => [],
-  },
-  sorts: {
-    type: Array,
-    default: () => [],
-  },
-  data: {
-    type: Array,
-    default: () => [],
-  },
-  totalRecords: {
-    type: Number,
-  },
-  maxHeight: String,
-  stickyColumn: {
-    type: String,
-    validator(value) {
-      return ['left', 'right'].includes(value);
-    },
-  },
-  title: String,
-  isCustomRowShown: {
-    type: Boolean,
-    default: () => undefined,
-  },
-  isTableHeaderHidden: {
-    type: Boolean,
-    default: () => undefined,
-  },
+const emit = defineEmits<{
+  'sort': [value: TableEvents[TableEvent.SORT]],
+}>();
+
+interface TableProps {
+  columns?: TableHeadColumn[],
+  sorts?: TableHeadSort[],
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  data?: any[],
+  totalRecords?: number,
+  maxHeight?: string,
+  stickyColumn?: StickyColumnPosition,
+  title?: string,
+  isCustomRowShown?: boolean,
+  isTableHeaderHidden?: boolean
+}
+
+const props = withDefaults(defineProps<TableProps>(), {
+  columns: () => [],
+  sorts: () => [],
+  data: () => [],
+  isCustomRowShown: undefined,
+  isTableHeaderHidden: undefined,
 });
 
 const numberOfColumns = computed(() => (props.columns.length || (props.data[0] && props.data[0].length) || null));
-const maxHeightStyle = computed(() => (props.maxHeight ? { maxHeight: `${props.maxHeight}` } : null));
+const maxHeightStyle = computed(() => (props.maxHeight ? { maxHeight: `${props.maxHeight}` } : undefined));
 const canShowCustomRow = computed(() => (props.isCustomRowShown || (props.isCustomRowShown === undefined && hasSlot('default') && isInCustomRowThreshold.value)));
 const canShowTableHeader = computed(() => (props.isTableHeaderHidden === false || (props.isTableHeaderHidden === undefined && !canShowCustomRow.value)));
 
-function onSort(columnName) {
-  emit('sort', columnName);
+function onSort(column: TableHeadColumn) {
+  emit('sort', column);
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function onRowClick(rowData: { data: any, rowIndex: number }) {
+  if (attrs.onRowClick) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    attrs.onRowClick(rowData);
+  }
 }
 
-function getColumnWidth(column) {
+function getColumnWidth(column: TableHeadColumn) {
   if (column && (column.maxWidth || column.minWidth)) {
     return { maxWidth: column.maxWidth, minWidth: column.minWidth };
   }
```

Comment:
We need to check if column exists before accessing those properties as because it's possible to have 0 columns (kinda weird to me tbh, but we have tests for that...)